lib/msf/core/exploit_driver.rb

```diff
@@ -215,7 +215,7 @@ def job_run_proc(ctx)
 
       # Build a user-friendly error message
       msg = "#{e}"
-      unless e.class == Msf::Exploit::Failed
+      unless e == Msf::Exploit::Failed
```

Comment:
Gonna test this to see exactly what the problem is or isn't. I could be wrong. :)